EarTrumpet/UI/Helpers/ShellNotifyIcon.cs

```diff
@@ -250,33 +251,41 @@ private void OnDelayedIconCheckForUpdate()
 
         public void ShowContextMenu(IEnumerable itemsSource)
         {
-            Trace.WriteLine("ShellNotifyIcon ShowContextMenu");
-            var contextMenu = new ContextMenu
+            if (!_isContextMenuOpen)
             {
-                FlowDirection = SystemSettings.IsRTL ? FlowDirection.RightToLeft : FlowDirection.LeftToRight,
-                StaysOpen = true,
-                ItemsSource = itemsSource,
-            };
-            Themes.Options.SetSource(contextMenu, Themes.Options.SourceKind.System);
-            contextMenu.PreviewKeyDown += (_, e) =>
-            {
-                if (e.Key == Key.Escape)
+                Trace.WriteLine("ShellNotifyIcon ShowContextMenu");
+                var contextMenu = new ContextMenu
                 {
-                    SetFocus();
-                }
-            };
-            contextMenu.Opened += (_, __) =>
-            {
-                Trace.WriteLine("ShellNotifyIcon ContextMenu.Opened");
+                    FlowDirection = SystemSettings.IsRTL ? FlowDirection.RightToLeft : FlowDirection.LeftToRight,
+                    StaysOpen = true,
+                    ItemsSource = itemsSource,
+                };
+                Themes.Options.SetSource(contextMenu, Themes.Options.SourceKind.System);
+                contextMenu.PreviewKeyDown += (_, e) =>
+                {
+                    if (e.Key == Key.Escape)
+                    {
+                        SetFocus();
+                    }
+                };
+                contextMenu.Opened += (_, __) =>
+                {
+                    Trace.WriteLine("ShellNotifyIcon ContextMenu.Opened");
                 // Workaround: The framework expects there to already be a WPF window open and thus fails to take focus.
                 User32.SetForegroundWindow(((HwndSource)HwndSource.FromVisual(contextMenu)).Handle);
-                contextMenu.Focus();
-                contextMenu.StaysOpen = false;
+                    contextMenu.Focus();
+                    contextMenu.StaysOpen = false;
                 // Disable only the exit animation.
                 ((Popup)contextMenu.Parent).PopupAnimation = PopupAnimation.None;
-            };
-            contextMenu.Closed += (_, __) => Trace.WriteLine("ShellNotifyIcon ContextMenu.Closed"); ;
-            contextMenu.IsOpen = true;
+                };
+                contextMenu.Closed += (_, __) =>
+                {
+                    Trace.WriteLine("ShellNotifyIcon ContextMenu.Closed");
+                    _isContextMenuOpen = false;
+                };
+                contextMenu.IsOpen = true;
+                _isContextMenuOpen = true;
```

Comment:
There may be a problem here due to how WPF does synchronously some Win32 window creation/setup stuff.  IsOpen=true will create the top level window to host the popup, and then return to caller.

If we did get multiple messages from the shell, we might process those messages during "IsOpen = true" because it is going to show the window and pump messages on the current thread relating to creation of that window.  Since the _isContextMenuOpen=true line is afterward, we could enter this code twice and create two ContextMenu objects.

I think it's worth avoiding this by putting `_isContextMenuOpen = true;` just after the `if` on line 254.